File: explorer/src/main/java/com/duongame/archive/RarFile.java
package com.duongame.archive;

import java.util.ArrayList;

/**
 * Created by namjungsoo on 2018-01-23.
 */

public class RarFile implements IArchiveFile {
    Unrar rar;

    public RarFile(String rarPath) {
        rar = new Unrar(rarPath);
    }

    @Override
    public ArrayList<ArchiveHeader> getHeaders() {
        ArrayList<UnrarHeader> headers = rar.getHeaders();
        ArrayList<ArchiveHeader> newHeaders = new ArrayList<>();

        for (UnrarHeader header : headers) {
            newHeaders.add(new ArchiveHeader(header.fileName, header.size));
        }
        return newHeaders;
    }

    @Override
    public boolean extractFile(String fileName, String destPath) {
        return rar.extractFile(fileName, destPath, null);
    }

    @Override
    public boolean extractAll(String destPath) {
        return rar.extractAll(destPath, null);
    }

    @Override
    public void destroy() {

    }
}

File: Explorer/src/main/java/com/duongame/archive/RarFile.kt
package com.duongame.archive

import java.util.*

/**
 * Created by namjungsoo on 2018-01-23.
 */
class RarFile(rarPath: String?) : ArchiveFile {
    var rar: Unrar = Unrar(rarPath)
    override val headers: ArrayList<ArchiveHeader>?
        get() {
            val headers = rar.headers ?: return null
            val newHeaders = ArrayList<ArchiveHeader>()
            for (header in headers) {
                newHeaders.add(ArchiveHeader(header.fileName, header.size))
            }
            return newHeaders
        }

    override fun extractFile(fileName: String, destPath: String): Boolean {
        return rar.extractFile(fileName, destPath, null)
    }

    override fun extractAll(destPath: String): Boolean {
        return rar.extractAll(destPath, null)
    }

    override fun destroy() {}
    override val isEncryped: Boolean
        get() = false

    override fun setPassword(password: String) {}
    override fun setFileNameCharset(charset: String) {}

}